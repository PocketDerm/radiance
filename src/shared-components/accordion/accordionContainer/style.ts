import styled from '@emotion/styled';
import { BOX_SHADOWS, BREAKPOINTS, COLORS } from 'src/constants';

const Container = styled.div<{ borderRadius?: string }>`
  border-radius: ${({ borderRadius }) => borderRadius || 'inherit'};
  box-shadow: ${BOX_SHADOWS.boxShadow4};
  background-color: ${COLORS.white};
  max-width: ${BREAKPOINTS.md}px;
`;

export default { Container };
