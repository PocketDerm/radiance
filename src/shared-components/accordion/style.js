import styled from '@emotion/styled';
import { css } from '@emotion/core';

import {
  ANIMATION,
  BOX_SHADOWS,
  BREAKPOINTS,
  COLORS,
  SPACER,
} from '../../constants';

const border = `1px solid ${COLORS.border}`;

export const Container = styled.div`
  box-shadow: ${BOX_SHADOWS.boxShadow4};
  background-color: ${COLORS.white};
  max-width: ${BREAKPOINTS.md}px;
`;

export const Content = styled.div`
  padding: ${SPACER.medium};
  width: 100%;
`;

export const ExpansionWrapper = styled.div`
  max-height: ${({ contentHeight }) => contentHeight};
  overflow: hidden;
  transition: max-height ${ANIMATION.defaultTiming} ease-in-out;
`;

const getBorderStyle = isOpen => css`
  border: ${border};

  ${ExpansionWrapper} {
    ${isOpen && `border-top: ${border};`};
  }
`;

export const AccordionBox = styled.div`
  ${({ noBorder, isOpen }) => !noBorder && getBorderStyle(isOpen)};
  width: 100%;

  &:not(:last-child) {
    border-bottom: none;
  }

  ${({ disabled }) =>
    disabled &&
    `
    opacity: 0.4;
    background-color: ${COLORS.disabled};
    border-color: ${COLORS.purple30};
  `};
`;

export const ArrowWrapper = styled.div`
  display: flex;
  align-items: center;
  padding: 0 ${SPACER.medium};
`;

export const TitleWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
`;

export const Truncate = styled.div`
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
`;
