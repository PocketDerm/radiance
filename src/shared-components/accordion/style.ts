import styled from '@emotion/styled';
import { css } from '@emotion/core';
import {
  ANIMATION,
  BREAKPOINTS,
  BOX_SHADOWS,
  COLORS,
  SPACER,
} from 'src/constants';

const border = `1px solid ${COLORS.border}`;

const getTopBorderRadius = (borderRadius: string) => ({
  borderTopLeftRadius: borderRadius,
  borderTopRightRadius: borderRadius,
});

const getBottomBorderRadius = (borderRadius: string) => ({
  borderBottomLeftRadius: borderRadius,
  borderBottomRightRadius: borderRadius,
});

export const Content = styled.div`
  padding: ${SPACER.medium};
  width: 100%;
`;

export const ExpansionWrapper = styled.div<{ contentHeight: string }>`
  max-height: ${({ contentHeight }) => contentHeight};
  overflow: hidden;
  transition: max-height ${ANIMATION.defaultTiming} ease-in-out;
`;

const getBorderStyle = (isOpen: boolean) => css`
  border: ${border};

  ${ExpansionWrapper} {
    ${isOpen && `border-top: ${border};`};
  }
`;

export const AccordionBox = styled.div<{
  noBorder: boolean;
  isOpen: boolean;
  disabled: boolean;
}>`
  ${({ noBorder, isOpen }) => (!noBorder ? getBorderStyle(isOpen) : '')};

  position: relative;

  width: 100%;

  &:not(:last-child) {
    border-bottom: none;
  }

  ${({ disabled }) =>
    disabled
      ? `
    opacity: 0.4;
    background-color: ${COLORS.disabled};
    border-color: ${COLORS.purple30};
  `
      : ''};
`;

export const ArrowWrapper = styled.div<{ rightAlign: boolean }>`
  display: flex;
  align-items: center;

  ${({ rightAlign }) =>
    rightAlign
      ? `padding-left: ${SPACER.medium};`
      : `padding: 0 ${SPACER.medium};`};
`;

export const TitleWrapper = styled.div<{
  borderRadius: string;
  disabled: boolean;
  isOpen: boolean;
}>`
  display: flex;
  justify-content: space-between;
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
  &:focus {
    outline: none;
    box-shadow: ${BOX_SHADOWS.focusSecondary};
  }

  div:last-of-type & {
    &:focus {
      ${({ borderRadius, isOpen }) => {
    const {
      borderBottomLeftRadius,
      borderBottomRightRadius,
    } = getBottomBorderRadius(borderRadius);

    return (
      !isOpen &&
          `
        border-bottom-left-radius: ${borderBottomLeftRadius}; 
        border-bottom-right-radius: ${borderBottomRightRadius};
        `
    );
  }}
    }
  }
`;

export const Truncate = styled.div`
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 100%;
`;

/**
 * borderRadius must match borderRadius passed to main <Accordion />
 * component if opting out of default values.
 */
export const Container = styled.div<{ borderRadius?: string }>`
  box-shadow: ${BOX_SHADOWS.clickable};
  background-color: ${COLORS.white};
  max-width: ${BREAKPOINTS.md}px;

  ${({ borderRadius = '4px' }) => {
    const { borderTopLeftRadius, borderTopRightRadius } = getTopBorderRadius(
      borderRadius,
    );
    const {
      borderBottomLeftRadius,
      borderBottomRightRadius,
    } = getBottomBorderRadius(borderRadius);

    return `
    > div:first-of-type {
      border-top-left-radius: ${borderTopLeftRadius};
      border-top-right-radius: ${borderTopRightRadius};

      ${TitleWrapper} {
        border-top-left-radius: ${borderTopLeftRadius};
        border-top-right-radius: ${borderTopRightRadius};
      }

      ${AccordionBox} {
        border-top-left-radius: ${borderTopLeftRadius};
        border-top-right-radius: ${borderTopRightRadius};
      }
    }
  
    > div:last-of-type {
      border-bottom-left-radius: ${borderBottomLeftRadius};
      border-bottom-right-radius: ${borderBottomRightRadius};
  
      ${AccordionBox} {
        border-bottom-left-radius: ${borderBottomLeftRadius};
        border-bottom-right-radius: ${borderBottomRightRadius};
      }
    }
`;
  }}
`;
