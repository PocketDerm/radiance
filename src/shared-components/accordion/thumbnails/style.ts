import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { style as TYPOGRAPHY_STYLE } from '../../typography';
import { COLORS, SPACER } from '../../../constants';

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  margin-left: ${SPACER.large};
`;

const thumbnailBase = css`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 1.5rem;
  width: 1rem;
`;

export const ImageContainer = styled.div`
  ${thumbnailBase} overflow: hidden;
  height: 1.5rem;
  width: 1rem;

  &:not(:last-child) {
    margin-right: 4px;
  }
`;

export const ThumbnailImage = styled.img`
  min-height: 1.5rem;
  min-width: 1rem;
  object-fit: cover;
`;

export const MultiplesContainer = styled.div`
  ${thumbnailBase}
  background-color: ${COLORS.purple15};
`;

export const MultiplesText = styled.div`
  ${TYPOGRAPHY_STYLE.label};
  font-size: 10px;
`;
