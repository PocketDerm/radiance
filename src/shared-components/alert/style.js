import styled from '@emotion/styled';
import { css, keyframes } from '@emotion/core';

import {
  COLORS,
  MEDIA_QUERIES,
  SPACER,
  ANIMATION,
  TYPOGRAPHY_CONSTANTS,
} from '../../constants';

export const AlertsContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  position: fixed;
  top: ${SPACER.medium};
  left: 0;
  right: 0;
  z-index: 99999;

  ${MEDIA_QUERIES.mdUp} {
    left: auto;
    right: ${SPACER.medium};
  }
`;

const fadeInDesktop = keyframes`
  from { opacity: 0; transform: translate3d(24px, 0, 0); }
  to { opacity: 1; transform: translate3d(0, 0, 0); }
`;

const fadeInMobile = keyframes`
  from { opacity: 0; transform: translate3d(0, -24px, 0);  }
  to { opacity: 1; transform: translate3d(0, 0px, 0); }
`;

const defaultAlertStyles = css`
  background-color: ${COLORS.primary};
  box-shadow: 0px 8px 24px rgba(51, 46, 84, 0.05);
`;

const successAlertStyles = css`
  background-color: ${COLORS.success};
  box-shadow: 0px 8px 24px rgba(43, 110, 51, 0.05);
`;

const errorAlertStyles = css`
  background-color: ${COLORS.error};
  box-shadow: 0px 8px 24px rgba(189, 32, 15, 0.05);
`;

export const AlertContainer = styled.div`
  cursor: pointer;
  position: relative;
  margin: 0 auto ${SPACER.small};
  padding: 0;
  width: 327px;
  border-radius: ${SPACER.small};
  opacity: ${props => (props.exiting ? '0' : '1')};
  animation: ${fadeInMobile} ${ANIMATION.defaultTiming} 1;
  transition: ${ANIMATION.defaultTiming};
  transform: ${({ exiting }) =>
    exiting ? 'translate3d(0, -24px, 0)' : 'translate3d(0, 0, 0)'};

  ${({ alertType }) => {
    switch (alertType) {
      case 'danger':
        return errorAlertStyles;
      case 'error':
        return errorAlertStyles;
      case 'success':
        return successAlertStyles;
      default:
        return defaultAlertStyles;
    }
  }};

  ${MEDIA_QUERIES.mdUp} {
    animation: ${fadeInDesktop} ${ANIMATION.defaultTiming} 1;
    margin: 0 auto ${SPACER.medium};
    transform: ${({ exiting }) =>
      exiting ? 'translate3d(24px, 0, 0)' : 'translate3d(0, 0, 0)'};
  }
`;

export const MainContainer = styled.div`
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  align-items: flex-start;
  color: ${COLORS.white};
  font-size: ${TYPOGRAPHY_CONSTANTS.fontSize.caption};
  padding: ${SPACER.medium};
`;

export const ContentContainer = styled.div`
  margin: -3px 0 0 ${SPACER.medium};
  max-height: 48px;
`;

export const CtaContent = styled.div`
  padding: ${SPACER.medium};

  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  color: ${COLORS.white};
  font-size: ${TYPOGRAPHY_CONSTANTS.fontSize.caption};

  ${({ alertType }) => {
    switch (alertType) {
      case 'danger':
        return css`
          border-top: 1px solid ${COLORS.errorBorder};
        `;
      case 'error':
        return css`
          border-top: 1px solid ${COLORS.errorBorder};
        `;
      case 'success':
        return css`
          border-top: 1px solid ${COLORS.successBorder};
        `;
      default:
        return css`
          border-top: 1px solid ${COLORS.infoBorder};
        `;
    }
  }};
`;

export const IconContainer = styled.div`
  ${({ hasAvatar }) =>
    hasAvatar &&
    css`
      min-width: 32px;
    `};

  svg {
    height: ${SPACER.medium};
    width: ${SPACER.medium};
    fill: ${COLORS.white};
  }
`;
