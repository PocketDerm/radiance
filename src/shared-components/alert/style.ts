import styled from '@emotion/styled';
import { css, keyframes, SerializedStyles } from '@emotion/core';

import {
  COLORS,
  MEDIA_QUERIES,
  SPACER,
  ANIMATION,
  TYPOGRAPHY_CONSTANTS,
} from '../../constants';

export const AlertsContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  position: fixed;
  top: ${SPACER.medium};
  left: 0;
  right: 0;
  z-index: 99999;

  ${MEDIA_QUERIES.mdUp} {
    left: auto;
    right: ${SPACER.medium};
  }
`;

const fadeInDesktop = keyframes`
  from { opacity: 0; transform: translate3d(24px, 0, 0); }
  to { opacity: 1; transform: translate3d(0, 0, 0); }
`;

const fadeInMobile = keyframes`
  from { opacity: 0; transform: translate3d(0, -24px, 0);  }
  to { opacity: 1; transform: translate3d(0, 0px, 0); }
`;

const defaultAlertStyles = css`
  background-color: ${COLORS.primary};
  box-shadow: 0px 8px 24px rgba(51, 46, 84, 0.05);
`;

const successAlertStyles = css`
  background-color: ${COLORS.success};
  box-shadow: 0px 8px 24px rgba(43, 110, 51, 0.05);
`;

const errorAlertStyles = css`
  background-color: ${COLORS.error};
  box-shadow: 0px 8px 24px rgba(189, 32, 15, 0.05);
`;

export const AlertContainer = styled.div<{
  exiting: boolean;
  alertType: string;
}>`
  cursor: pointer;
  position: relative;
  margin: 0 auto ${SPACER.small};
  padding: 0;
  width: 327px;
  border-radius: ${SPACER.small};
  opacity: ${({ exiting }): string => (exiting ? '0' : '1')};
  animation: ${fadeInMobile} ${ANIMATION.defaultTiming} 1;
  transition: ${ANIMATION.defaultTiming};
  transform: ${({ exiting }): string =>
    exiting ? 'translate3d(0, -24px, 0)' : 'translate3d(0, 0, 0)'};

  ${({ alertType }): SerializedStyles => {
    switch (alertType) {
      case 'danger':
        return errorAlertStyles;
      case 'error':
        return errorAlertStyles;
      case 'success':
        return successAlertStyles;
      default:
        return defaultAlertStyles;
    }
  }};

  ${MEDIA_QUERIES.mdUp} {
    animation: ${fadeInDesktop} ${ANIMATION.defaultTiming} 1;
    margin: 0 auto ${SPACER.medium};
    transform: ${({ exiting }): string =>
    exiting ? 'translate3d(24px, 0, 0)' : 'translate3d(0, 0, 0)'};
  }
`;

export const MainContainer = styled.div`
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  align-items: flex-start;
  color: ${COLORS.white};
  font-size: ${TYPOGRAPHY_CONSTANTS.fontSize.caption};
  padding: ${SPACER.medium};
`;

export const ContentContainer = styled.div<{ truncateText: boolean }>`
  margin: -3px 0 0 ${SPACER.medium};
  max-height: ${({ truncateText }): string => (truncateText ? '48px' : 'none')};
`;

export const CtaContent = styled.div`
  padding: ${SPACER.medium};

  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  color: ${COLORS.white};
  font-size: ${TYPOGRAPHY_CONSTANTS.fontSize.caption};
  border-top: 1px solid rgba(255, 255, 255, 0.1);
`;

export const IconContainer = styled.div<{ hasAvatar: boolean }>`
  ${({ hasAvatar }): SerializedStyles | false =>
    hasAvatar &&
    css`
      min-width: 32px;
    `};

  svg {
    height: ${SPACER.medium};
    width: ${SPACER.medium};
    fill: ${COLORS.white};
  }
`;
