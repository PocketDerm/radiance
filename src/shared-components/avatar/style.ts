import styled from '@emotion/styled';

const SIZES = {
  small: 32,
  medium: 48,
  large: 80,
};

const determineSize = ({avatarSize}: {
  avatarSize: 'small' | 'medium' | 'large';
}) => `
  height: ${SIZES[avatarSize]}px;
  width: ${SIZES[avatarSize]}px;
`;

const AvatarImage = styled.img<{
  avatarSize: 'small' | 'medium' | 'large';
}>`
  ${determineSize}
  background-color: ${({ theme }) => theme.COLORS.secondary};
  border-radius: 50%;
  overflow: hidden;
`;

export default {
  AvatarImage,
};
