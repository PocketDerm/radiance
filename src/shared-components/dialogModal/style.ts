import { css } from '@emotion/react';

import {
  Colors,
  MEDIA_QUERIES,
  SPACER,
  ThemeType,
  Z_SCALE,
} from '../../constants';

const overlay = (theme: ThemeType) => css`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: ${Z_SCALE.modal};
  background-color: ${theme.COLORS.overlay};
  transition: opacity 250ms cubic-bezier(0.075, 0.82, 0.165, 1);

  &.entering,
  &.exiting,
  &.exited {
    opacity: 0;
  }

  &.entered {
    opacity: 1;
  }

  ${MEDIA_QUERIES.mdUp} {
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
  }
`;

const paragraph = css`
  margin-bottom: ${SPACER.large};

  &:last-of-type {
    margin-bottom: ${SPACER.xlarge};
  }
`;

const modalContainer = (
  backgroundColor: Colors['background'] | Colors['white'],
) => (theme: ThemeType) => css`
  width: 100%;
  margin: 0 auto;
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  border-top-left-radius: ${theme.BORDER_RADIUS.large};
  border-top-right-radius: ${theme.BORDER_RADIUS.large};
  box-shadow: ${theme.BOX_SHADOWS.modal};
  background: ${backgroundColor};
  padding: ${SPACER.x4large} ${SPACER.large} ${SPACER.xlarge};
  overflow-y: auto;
  max-height: 100%;

  transition: transform 250ms cubic-bezier(0.075, 0.82, 0.165, 1);

  &.entered {
    transform: translateY(0%);
  }

  &.entering,
  &.exiting,
  &.exited {
    transform: translateY(100%);
  }

  ${MEDIA_QUERIES.smUp} {
    max-height: 700px;
  }

  ${MEDIA_QUERIES.mdUp} {
    position: relative;
    width: 456px;
    border-radius: ${theme.BORDER_RADIUS.medium};
    padding: ${SPACER.x4large};

    &.entering,
    &.exiting,
    &.exited {
      transform: translateY(40%);
    }
  }
`;

const modalTitle = css`
  margin-bottom: ${SPACER.small};
`;

const crossIconContainer = (
  backgroundColor: Colors['background'] | Colors['white'],
) => (theme: ThemeType) => css`
  position: absolute;
  top: 8px;
  right: 12px;
  z-index: ${Z_SCALE.e2000};
  width: 40px;
  height: 40px;
  border-radius: 50%;
  background: ${backgroundColor};
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  align-items: center;
  pointer-events: auto;
  cursor: pointer;

  ${MEDIA_QUERIES.mdUp} {
    top: 16px;
    right: 16px;
  }
  &:focus {
    outline: none;
    box-shadow: ${theme.BOX_SHADOWS.focus};
  }
`;

export default {
  crossIconContainer,
  modalContainer,
  modalTitle,
  overlay,
  paragraph,
};
