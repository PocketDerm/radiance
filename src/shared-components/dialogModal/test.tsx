import React from 'react';
import { mount } from 'enzyme';

import { ModalTitle } from './style';

import DialogModal from './index';

const modalTitle = 'Dialog Modal Title';
const modalBody = 'Dialog Modal Children Content';

describe('<DialogModal />', () => {
  it('render children content correctly', () => {
    const wrapper = mount(
      <DialogModal title={modalTitle}>
        <div>{modalBody}</div>
      </DialogModal>,
    );

    expect(wrapper.find(ModalTitle).text()).toBe(modalTitle);
    expect(wrapper.text().includes(modalBody)).toBeTruthy();
  });
});
