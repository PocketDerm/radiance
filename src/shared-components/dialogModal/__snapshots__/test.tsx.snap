// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogModal /> renders dialog modal with custom color 1`] = `
.css-ncv863-modalContainer.entering,
.css-ncv863-modalContainer.exiting,
.css-ncv863-modalContainer.exited {
  -webkit-transform: translateY(100%);
  -moz-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

@media (min-width: 768px) {
  .css-ncv863-modalContainer {
    position: relative;
    width: 456px;
    border-radius: 8px;
    padding: 3.5rem;
  }

  .css-ncv863-modalContainer.entering,
  .css-ncv863-modalContainer.exiting,
  .css-ncv863-modalContainer.exited {
    -webkit-transform: translateY(40%);
    -moz-transform: translateY(40%);
    -ms-transform: translateY(40%);
    transform: translateY(40%);
  }
}

.emotion-0 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 999999;
  background-color: rgba(45, 45, 48, 0.7);
  -webkit-transition: opacity 250ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: opacity 250ms cubic-bezier(0.075, 0.82, 0.165, 1);
}

.emotion-0.entering,
.emotion-0.exiting,
.emotion-0.exited {
  opacity: 0;
}

.emotion-0.entered {
  opacity: 1;
}

@media (min-width: 768px) {
  .emotion-0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-flex-flow: row nowrap;
    -webkit-flex-flow: row nowrap;
    -ms-flex-flow: row nowrap;
    flex-flow: row nowrap;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

.emotion-1 {
  width: 100%;
  margin: 0 auto;
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  border-top-left-radius: 32px;
  border-top-right-radius: 32px;
  box-shadow: 0px -8px 24px rgba(52, 51, 82, 0.05);
  background: #F8F8FA;
  padding: 3.5rem 1.5rem 2rem;
  overflow-y: auto;
  max-height: 100%;
  -webkit-transition: -webkit-transform 250ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: transform 250ms cubic-bezier(0.075, 0.82, 0.165, 1);
}

.emotion-1.entered {
  -webkit-transform: translateY(0%);
  -moz-transform: translateY(0%);
  -ms-transform: translateY(0%);
  transform: translateY(0%);
}

.emotion-1.entering,
.emotion-1.exiting,
.emotion-1.exited {
  -webkit-transform: translateY(100%);
  -moz-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

@media (min-width: 576px) {
  .emotion-1 {
    max-height: 700px;
  }
}

@media (min-width: 768px) {
  .emotion-1 {
    position: relative;
    width: 456px;
    border-radius: 8px;
    padding: 3.5rem;
  }

  .emotion-1.entering,
  .emotion-1.exiting,
  .emotion-1.exited {
    -webkit-transform: translateY(40%);
    -moz-transform: translateY(40%);
    -ms-transform: translateY(40%);
    transform: translateY(40%);
  }
}

.emotion-2 {
  margin-bottom: 1.5rem;
}

.emotion-2:last-of-type {
  margin-bottom: 2rem;
}

<div
  class="entering emotion-0"
>
  <span
    hidden=""
  />
  <div
    class="entering emotion-1"
  >
    <div>
      <p
        class="emotion-2"
      >
        Dialog Modal Children Content
      </p>
    </div>
  </div>
  <span
    hidden=""
  />
</div>
`;
